### DISCUSSION TITLE
Integration of jupynium

### DISCUSSION THREAD

**Discussion by statquant** [quarto-dev/quarto-nvim#52] [Ideas]
Hello,
thanks for this nvim addin it’s really great.
I see that quarto-nvim takes advantages of many other addins so I wanted to bring your attention to https://github.com/kiyoon/jupynium.nvim
We are using it and found it to be a major productivity boost.
This idea is that you can send code to jupyter for visualisation. The advantage over preview arise when some cells require long execution and even loading back the cache take a lot of time. This is always the case in our field.
You could even push it to send as well in a R console in a tmux pane (most of us dislike working in a notebook cell) and essentially keep the notebook session 'cleaner'

Hope you find jupynium worth a look

**Comment by jmbuhr** [comment]
Thanks, this looks really cool! This attaches to a plain python file with code chunk marker comments, right? So if I we add those markers to the synchronized otter buffer of https://github.com/jmbuhr/otter.nvim, you could attach it to this hidden buffer, which would then keep both things in sync.

A quick word from educator-me: If your code cells take a long time to execute and even caching takes too long, it is most likely a sign that you are better off refactoring this part into a script or function and track it with a proper workflow/dependency tool such as `Make` or in the R specific case `targets` (really cool package by the way ;) ).

**Reply by statquant** [reply]
Hello, I was thinking that requesting the jupynium maintainer to add quarto format support or PR-ing it would probably be simple enough to avoid having to maintain the copy.

Thanks for your suggestion, I actually solved it through pre-chunk hooks that checks if the chunk has changed and exec : false if it does not.
Pb is that targets requires people to learn it.
